lighthouse-core/audits/first-meaningful-paint.js

```diff
@@ -181,6 +156,11 @@ class FirstMeaningfulPaint extends Audit {
           !!navigationStart && event.ts >= navigationStart.ts) {
         firstContentfulPaint = event;
       }
+      if (event.name === 'firstMeaningfulPaint' && event.args.frame === mainFrameID &&
+          !!navigationStart && event.ts >= navigationStart.ts) {
+        firstMeaningfulPaint = event;
+      }
+
       // COMPAT: frame property requires Chrome 52 (r390306)
       // https://codereview.chromium.org/1922823003
       if (event.name === 'FrameView::performLayout' &&
```

Comment:
Also you can remove the `disabled-by-default-blink.debug.layout` trace category

https://github.com/GoogleChrome/lighthouse/blob/1b6a53188a03c1218ad4e53f49ef4284c1195437/lighthouse-core/gather/driver.js#L52